### DISCUSSION TITLE
Halo使用CloudFlare以后登陆提示An expected CSRF token cannot be found

### DISCUSSION THREAD

**Discussion by kkqy** [halo-dev/halo#7040] [Q&A]
我把cloudflare的缓存关闭，然后各种功能都关闭了，还是提示An expected CSRF token cannot be found。
关掉cloudflare就正常了，我估计是cloudflare因为什么原因过滤了_csrf参数？

**Comment by kkqy** [comment]
解决了，是cloudflare缓存规则里的优先级问题

**Reply by reichou1989** [reply]
也遇到一樣的問題，麻煩告知也一下解決方法，是要設置什麽内容跳過緩存嗎

**Reply by kkqy** [reply]
> 也遇到一樣的問題，麻煩告知也一下解決方法，是要設置什麽内容跳過緩存嗎

只能让它缓存静态资源，比如后缀是jpg、js、css、png这种，不要让它以任何形式缓存任何post的地址。如果你缓存规则比较复杂，建议把所有缓存规则删除，确保没有问题之后，再逐步添加，确定是哪一条的问题。

**Reply by reichou1989** [reply]
> > 也遇到一樣的問題，麻煩告知也一下解決方法，是要設置什麽内容跳過緩存嗎
> 
> 只能让它缓存静态资源，比如后缀是jpg、js、css、png这种，不要让它以任何形式缓存任何post的地址。如果你缓存规则比较复杂，建议把所有缓存规则删除，确保没有问题之后，再逐步添加，确定是哪一条的问题。

嗯，參照了wp的方法跳過了一些COOKIE，還是一樣，就算能登陸，也會緩存登錄狀態。現在只緩存靜態資源了。效果還算可以。謝謝解答。
